pkg/asset/manifests/proxy.go

```diff
@@ -142,7 +142,9 @@ func createNoProxy(installConfig *installconfig.InstallConfig, network *Networki
 	}
 
 	for _, userValue := range strings.Split(installConfig.Config.Proxy.NoProxy, ",") {
-		set.Insert(userValue)
+		if userValue != "" {
```

Comment:
should we validate rather than silently ignore?

Reply:
we want to silently ignore the empty-string case.  I'm fine rejecting entries with a leading or trailing comma in validation if you think we should be doing that instead.

Reply:
which would be something like:

if len(installConfig.Config.Proxy.NoProxy) > 0 {
  for _, userValue := range strings.Split(installConfig.Config.Proxy.NoProxy, ",") {
    if userValue != "" {
      return validationError
    }
    addtoset(userValue)
  }
}
```

Reply:
more as to doing the validation here https://github.com/openshift/installer/blob/master/pkg/types/validation/installconfig.go

Reply:
> more as to doing the validation here https://github.com/openshift/installer/blob/master/pkg/types/validation/installconfig.go

So in that case I think we want what I have here to handle the empty string case, and then we can add follow-up work to forbid leading, trailing, or doubled commas.  Should I do that in this PR or in follow-up work?

Reply:
this PR should be fine

Reply:
Turns out we [already guard against this][1] in validation.  Proof:

```console
$ tail -n4 aws.yaml
proxy:
  noProxy: ',localhost'
  httpProxy: http://admin:admin@35.196.128.173:3128/
  httpsProxy: http://admin:admin@35.196.128.173:3128/
$ mkdir testing
$ cp aws.yaml testing/install-config.yaml
$ openshift-install --dir testing create manifests
FATAL failed to fetch Master Machines: failed to load asset "Install Config": invalid "install-config.yaml" file: NoProxy: Invalid value: "": must be a CIDR or domain, without wildcard characters and without leading or trailing dots ('.') 
```

[1]: https://github.com/openshift/installer/blob/caa9246bdca697fe3651e801d0ed4b1d27632139/pkg/types/validation/installconfig.go#L265-L274

Reply:
Oh nice, thanks to @jcpowermac :)

So I think we need to only loop when spec.noProxy != "" otherwise we should be set..

Reply:
`$ cat proxy/manifests/cluster-proxy-01-config.yaml | grep noProxy`
before change (note leading comma and quotes):  `noProxy: ',10.128.0.0/14,127.0.0.1,169.254.169.254,172.30.0.0/16,api-int.padillon.devcluster.openshift.com,api.padillon.devcluster.openshift.com,etcd-0.padillon.devcluster.openshift.com,etcd-1.padillon.devcluster.openshift.com,etcd-2.padillon.devcluster.openshift.com,localhost'`
with this change:  `noProxy: 10.128.0.0/14,127.0.0.1,169.254.169.254,172.30.0.0/16,api-int.padillon.devcluster.openshift.com,api.padillon.devcluster.openshift.com,etcd-0.padillon.devcluster.openshift.com,etcd-1.padillon.devcluster.openshift.com,etcd-2.padillon.devcluster.openshift.com,localhost`

Thanks for catching this!
/lgtm